### PR TITLE
Upgrade Playwright to v1.42

### PR DESCRIPTION
**Pull request by Mamaduka**
## What?
Upgrade Playwright to v1.41.2

## What's new?

https://github.com/microsoft/playwright/releases/tag/v1.41.0
https://github.com/microsoft/playwright/releases/tag/v1.42.0

## Testing Instructions
* Run any e2e test locally; the command should download new browsers and run the test successfully.
* CI checks are green.

### CODE REVIEW

File: packages/scripts/scripts/test-playwright.js [hunk 1]
```diff
@@ -28,20 +27,9 @@ const {
...
-			),
-			'install',
-		],
-		{
-			stdio: 'inherit',
-		}
-	);
+	const result = spawn( 'npx', [ 'playwright', 'install' ], {
+		stdio: 'inherit',
+	} );
```
Comment (Mamaduka) [inline packages/scripts/scripts/test-playwright.js:32]: See https://github.com/WordPress/gutenberg/pull/56594/files#r1408876091

File: test/e2e/specs/editor/various/block-switcher.spec.js [hunk 2]
```diff
@@ -131,9 +132,7 @@ test.describe( 'Block Switcher', () => {
 		await page.keyboard.press( 'ArrowUp' );
 		await page.keyboard.press( 'Enter' );
 		await page.keyboard.type( '## I am a header' );
-		await page.keyboard.down( 'Shift' );
-		await page.keyboard.press( 'ArrowUp' );
-		await page.keyboard.up( 'Shift' );
+		await pageUtils.pressKeys( 'primary+a', { times: 2 } );
```
Comment (Mamaduka) [inline test/e2e/specs/editor/various/block-switcher.spec.js:135]: I'm unsure if this is a bug in core or just an e2e test quirk, but when using `pageUtils.pressKeys( 'shift+ArrowUp' )` block toolbar isn't visible.

I can't reproduce the issue when testing manually.

## Screenshot
test-failed-1

**Comment by Mamaduka** [comment]
I see a problem with artifacts - https://github.com/WordPress/gutenberg/actions/runs/8044664244?pr=59339. We might need to update the configuration.

**Update:** It might be unrelated. I see same error on other PRs - https://github.com/WordPress/gutenberg/actions/runs/8040401372?pr=59350

## Screenshot
CleanShot 2024-02-26 at 11 26 36

**Comment by kevin940726** [comment]
They just released [v1.42.0](https://github.com/microsoft/playwright/releases/tag/v1.42.0), might be good to update to that as well 😆?

**Comment by Mamaduka** [comment]
I updated Playwright to the latest release.

**Update:** The e2e test failures seem to come from the update. I am unsure what has changed; I will look into it later.

**Comment by Mamaduka** [comment]
It looks like there's a regression with Playwright v1.42 and chained locators (which the project uses heavily). Reported it upstream - https://github.com/microsoft/playwright/issues/29760.

**Comment by Mamaduka** [comment]
Okay, I rebased on more time and will merge if the checks are green.

**Pull request merged**